boards/xtensa/esp32/esp32_defconfig

```diff
@@ -32,3 +32,8 @@ CONFIG_I2C_1=y
 CONFIG_CLOCK_CONTROL=y
 
 CONFIG_BOOTLOADER_ESP_IDF=y
+CONFIG_HEAP_MEM_POOL_SIZE=98304
+CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE=4096
+CONFIG_POSIX_API=y
+CONFIG_THREAD_CUSTOM_DATA=y
```

Comment:
Are all these setting necessary?

Reply:
Not all settings are necessary. I've kept `CONFIG_HEAP_MEM_POOL_SIZE=98304` - which is necessary to reference `k_calloc` - and removed the remaining.

Reply:
> Not all settings are necessary. I've kept `CONFIG_HEAP_MEM_POOL_SIZE=98304` - which is necessary to reference `k_calloc` - and removed the remaining.

@chrta  I have moved `CONFIG_HEAP_MEM_POOL_SIZE` from `esp32_dev_config` to both `tests/drivers/counter/counter_basic_api/boards/esp32.conf` and `samples/driver/counter/alarm/boards/esp32.conf` since it is a project specific configuration. The reserved heap was too much and was also updated.